Nasal/light-manager.nas

```diff
@@ -0,0 +1,303 @@
+        delta_y = -(apos.lon() - vpos.lon()) * me.lon_to_m;
+        delta_z = apos.alt() - vpos.alt();
+
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-x-m[3]", delta_x);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-y-m[3]", delta_y);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/eyerel-z-m[3]", delta_z);
+
+        settimer(func me.update(), 0.0);
+    },   
+
+    light1_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r", me.light1_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g", me.light1_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b", me.light1_b);
+    },
+
+    light1_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b", 0.0);
+    },
+
+    light2_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[1]", me.light2_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[1]", me.light2_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[1]", me.light2_b);
+    },
+
+    light2_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[1]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[1]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[1]", 0.0);
+    },
+
+    light3_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[2]", me.light3_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[2]", me.light3_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[2]", me.light3_b);
+    },
+
+    light3_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[2]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[2]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[2]", 0.0);
+    },
+
+    light4_on: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[3]", me.light4_r);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[3]", me.light4_g);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[3]", me.light4_b);
+    },
+
+    light4_off: func {
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-r[3]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-g[3]", 0.0);
+        setprop("/sim/rendering/als-secondary-lights/lightspot/lightspot-b[3]", 0.0);
+    },
+    
+    enable_or_disable: func (enable, light_num) {
+        if (enable) {
```

Comment:
Instead of generating functions like this:

```javascript
var light1_on = lightspot_on_func(0);
```

You could write `lightspot_on_func(light_num)()` instead. Imagine the number of if statements if you have 20 or 100 instead of 4 lights.